lib/chef/resource/windows_pagefile.rb

```diff
@@ -102,20 +114,32 @@ class WindowsPagefile < Chef::Resource
       end
 
       action :delete, description: "Deletes the specified pagefile." do
-        validate_name
-        delete(new_resource.path) if exists?(new_resource.path)
+        pagefile = clarify_pagefile_name
+        delete(pagefile) if exists?(pagefile)
       end
 
       action_class do
         private
 
-        # make sure the provided name property matches the appropriate format
-        # we do this here and not in the property itself because if automatic_managed
-        # is set then this validation is not necessary / doesn't make sense at all
-        def validate_name
-          return if /^.:.*.sys/.match?(new_resource.path)
+        # We are adding support for a number of possibilities for how users will express the drive and location they want the pagefile written to.
+        def clarify_pagefile_name
+          case new_resource.path
+          # user enters C, C:, C:\, C:\\
+          when /^[a-zA-Z]/
+            new_resource.path[0] + ":\\pagefile.sys"
+          # user enters C:\pagefile.sys OR c:\foo\bar\pagefile.sys as the path
+          when /^[a-zA-Z]:.*.sys/
+            new_resource.path
+          else
+            raise "#{new_resource.path} does not match the format DRIVE:\\path\\pagefile.sys for pagefiles. Example: C:\\pagefile.sys"
```

Comment:
you might wanna say "does not match the format DRIVE:\\ or DRIVE:\\path\\pagefile.sys for pagefiles."

Reply:
ok, tnanks

Reply:
done